### PR TITLE
Fix progress arrow for Workbench

### LINKED ISSUE #28266: Progress Arrow on wrong line when executing a section of a script
Speak to @martyngigg 
### Expected behavior
Progress Arrow should show the lines it is executing 
### Actual Behavior
Doesn't always get it right, such as in the case detailed below
### Steps to reproduce
In a script, highlight one or many lines and execute them, the progress arrow displays at the top of the script (eg. If you ran one line in the middle of your script (could be line 5), it will display a green arrow on line 1)
### Platforms affected
Workbench 4.2.29022020

### PR DESCRIPTION
**Pull request by DavidFair**
**Description of work.**
Fixes the progress arrow which previously only pointed at the first
line, irrespective of selection. Additionally removes a method which is
completely unused in the code-base, but looks like it would do a similar
thing as it's misleading.

A struct is used in case we need to add additional fields, and for the
extra readability when we dereference the map iterator. This avoids a
iter->value.at<0> in favour of having named fields of a struct.

**To test:**
- Run the following multiple times 
```python
# import mantid algorithms, numpy and matplotlib
from mantid.simpleapi import *
import matplotlib.pyplot as plt
import numpy as np

import time

time.sleep(2)
time.sleep(2)
time.sleep(2)
time.sleep(2)
time.sleep(2)
```

- Try selecting different subsets the script (e.g. the last 3 lines)
- Check the progress arrow always starts at the right location

Fixes #28266 . 

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: docs/source/release/v5.0.0/mantidworkbench.rst [hunk 1]
```diff
@@ -114,5 +114,6 @@ Bugfixes
 - Fixed bug that caused an error if a MDHistoWorkspace was plotted and a user attempted to open a context menu.
 - Fixed a bug which caused graphic scaling issues when the double-click menu was used to set an axis as log-scaled.
 - Fixed a bug where the colorbar in the instrument view would sometimes have no markers if the scale was set to SymmetricLog10.
+- Fixed a bug which caused the script progress arrow to always start on line 0, regardless of the users selected code.
```
Comment (martyngigg) [inline docs/source/release/v5.0.0/mantidworkbench.rst]: This was actually a regression in this dev cycle so we shouldn't need the release note.

File: qt/python/mantidqt/widgets/codeeditor/interpreter.py [hunk 2]
```diff
@@ -362,8 +364,3 @@ def _finish(self, success, task_result):
 
     def _create_status_msg(self, status, timestamp, elapsed_time):
         return IDLE_STATUS_MSG + ' ' + LAST_JOB_MSG_TEMPLATE.format(status, timestamp, elapsed_time)
-
-    def _on_progress_update(self, lineno):
-        """Update progress on the view taking into account if a selection of code is
```
Comment (martyngigg) [inline qt/python/mantidqt/widgets/codeeditor/interpreter.py]: Thanks for removing this. We missed it when this was changed just before code freeze.

File: qt/python/mantidqt/widgets/codeeditor/test/test_execution.py [hunk 3]
```diff
@@ -87,16 +87,29 @@ def test_script_dir_added_to_path_on_execution(self):
...
 
     def test_script_dir_removed_from_path_after_execution(self):
         code = "import sys; syspath = sys.path"
         test_filename = '/path/to/script/called/script.py'
         executor = PythonCodeExecution()
-        executor.execute(code, filename=test_filename)
+        executor.execute(code, filename=test_filename, line_offset=0)
         self.assertNotIn('/path/to/script/called', sys.path)
 
+    def test_line_offset_forwarded_on(self):
```
Comment (martyngigg) [inline qt/python/mantidqt/widgets/codeeditor/test/test_execution.py]: Good test :+1:

**Review by martyngigg — CHANGES_REQUESTED** [review]
This looks good and the progress bar is now in the correct place regardless of the selection.

Just a minor point about the release notes and then we are good.

File: docs/source/release/v5.0.0/mantidworkbench.rst (hunk 1 above)
Comment (DavidFair) [reply docs/source/release/v5.0.0/mantidworkbench.rst]: Okay I've removed it

**Review by martyngigg — APPROVED** [review]

**Comment by DavidFair** [comment]
Just an FYI: I won't be in on Monday so feel free to make any changes if required 
Thanks for the quick review 🥇

**Comment by martyngigg** [comment]
Looks good to me. Thanks for the quick fix.

**Pull request merged**